### PR TITLE
R4R: Finish the refund and delete deposit

### PR DESCRIPTION
**Pull request by MrXJC**

### CODE REVIEW

File: modules/gov/handler.go [hunk 1]
```diff
@@ -294,15 +296,15 @@ func EndBlocker(ctx sdk.Context, keeper Keeper) (resTags sdk.Tags) {
 						val.GetConsAddr(),
 						ctx.BlockHeight(),
 						val.GetPower().RoundInt64(),
-						GetTallyingProcedure(ctx).GovernancePenalty)
+						activeProposal.GetTallyingProcedure().GovernancePenalty)
```
Comment (wukongcheng) [inline modules/gov/handler.go]: do not need to store these params since them will not change

File: modules/gov/keeper.go [hunk 2]
```diff
@@ -9,14 +9,15 @@ import (
...
+	stakeTypes "github.com/irisnet/irishub/modules/stake/types"
 	"github.com/tendermint/tendermint/crypto"
 	"time"
 )
 
 // nolint
 var (
 	DepositedCoinsAccAddr     = sdk.AccAddress(crypto.AddressHash([]byte("govDepositedCoins")))
-	BurnedDepositCoinsAccAddr = sdk.AccAddress(crypto.AddressHash([]byte("govBurnedDepositCoins")))
+	BurnDeposit               = sdk.NewCoin(stakeTypes.StakeDenom, sdk.NewIntWithDecimal(2, 18))// 200iris
```
Comment (haifengxi) [inline modules/gov/keeper.go]: 200 iris = 2*10^20 iris-atto

File: modules/gov/keeper.go [hunk 3]
```diff
@@ -472,20 +474,41 @@ func (keeper Keeper) GetDeposits(ctx sdk.Context, proposalID uint64) sdk.Iterato
...
 		deposit := &govtypes.Deposit{}
 		keeper.cdc.MustUnmarshalBinaryLengthPrefixed(depositsIterator.Value(), deposit)
+		deposits = append(deposits,deposit)
+		depositSum = depositSum.Plus(deposit.Amount)
+		store.Delete(depositsIterator.Key())
+	}
+
+
+	rate := sdk.NewDecFromInt(BurnDeposit.Amount).Quo(sdk.NewDecFromInt(depositSum.AmountOf(stakeTypes.StakeDenom)))
+	if rate.IsNegative(){
```
Comment (wukongcheng) [inline modules/gov/keeper.go]: delete

File: modules/gov/keeper.go [hunk 4]
```diff
@@ -472,20 +474,41 @@ func (keeper Keeper) GetDeposits(ctx sdk.Context, proposalID uint64) sdk.Iterato
...
+		panic("Burn deposit must be less than deposit sum .")
+	}
+
+	burnAmountDec := sdk.NewDecWithPrec(0,0)
+	for _, deposit := range deposits {
+		AmountDec := sdk.NewDecFromInt(deposit.Amount.AmountOf(stakeTypes.StakeDenom))
+		burnAmountDec = burnAmountDec.Add(AmountDec.Mul(rate))
+		leftAmountDec := AmountDec.Sub(AmountDec.Mul(rate))
+		deposit.Amount = sdk.Coins{sdk.NewCoin(stakeTypes.StakeDenom,leftAmountDec.RoundInt())}
```
Comment (chengwenxi) [inline modules/gov/keeper.go]: 1. Why only StakeDenom ?
2. RoundInt()?

**Pull request merged**